### PR TITLE
arm/stm32h7x3x_rcc.c: Add External Power Supply option to stm32h7x3x …

### PR DESCRIPTION
**Pull request by TOTON95**
Targets: Devices using `arm/stm32h7x3x_rcc.c`

## Summary

This adds the option of using an external power supply connected directly to the VCAP pin.

This is necessary if the board designer decides to power the uC through an external power supply, which can help to decrease the temperature, have a stable current delivery, etc.

I kept this modification as simple as possible, so further combinations could be applied as shown in the reference manual RM0399, pages 275 and 329. It also is adapted to requirements shown in RM0433, pages 260 and 307.

## Impact

This should have minimal or lack of any impact on present targets, as it's using a new setting built around the existing power options structure, keeping the current configurations untouched.

## Testing

- Checked correct compilation against `nucleo-h743zi` and `stm32h747i-disco` targets, modifying the configuration of settings involved.
- Tested in HW with downstream (PX4) using 3DR Control Zero H7 OEM rev G.

Feedback is always welcome!

### CODE REVIEW

File: arch/arm/src/stm32h7/Kconfig [hunk 1]
```diff
@@ -293,6 +293,10 @@ config STM32H7_PWR_DIRECT_SMPS_SUPPLY
 	depends on STM32H7_HAVE_SMPS
 	default n
 
+config STM32H7_PWR_EXTERNAL_SOURCE_SUPPLY
+	bool "Use external source as power supply"
+	default n
+
```
Comment (davids5) [inline arch/arm/src/stm32h7/Kconfig:299]: Please add help 'Select this option for using an external power supply connected directly to the VCAP pin.'

**Review by davids5 — CHANGES_REQUESTED** [review]

**Review by acassis — APPROVED** [review]

File: arch/arm/src/stm32h7/Kconfig (hunk 1 above)
Comment (TOTON95) [reply arch/arm/src/stm32h7/Kconfig:301]: @davids5 No problem, it's now added.

**Comment by TOTON95** [comment]
@xiaoxiang781216 My bad, I just squashed it.

File: arch/arm/src/stm32h7/Kconfig [hunk 2]
```diff
@@ -296,6 +296,8 @@ config STM32H7_PWR_DIRECT_SMPS_SUPPLY
 config STM32H7_PWR_EXTERNAL_SOURCE_SUPPLY
 	bool "Use external source as power supply"
 	default n
+    ---help---
```
[the remark this answers is not in the document]
Comment (TOTON95) [reply arch/arm/src/stm32h7/Kconfig]: @acassis Fixed 🙌🏼

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by TOTON95** [comment]
@xiaoxiang781216 @acassis @davids5 Thank you!

**Pull request merged**